src/zope/testrunner/tb_format.py

```diff
@@ -20,9 +20,34 @@
 import zope.testrunner.feature
 
 
+try:
+    _iter_chain = traceback._iter_chain
+except AttributeError:
+    # Python 3.5
+    def _iter_chain(exc, custom_tb=None, seen=None):
+        if seen is None:
+            seen = set()
+        seen.add(exc)
+        its = []
+        cause = exc.__cause__
+        if cause is not None and cause not in seen:
+            its.append(_iter_chain(cause, None, seen))
+            its.append([(traceback._cause_message, None)])
+        else:
+            context = exc.__context__
+            if context is not None and context not in seen:
+                its.append(_iter_chain(context, None, seen))
+                its.append([(traceback._context_message, None)])
+        its.append([(exc, custom_tb or exc.__traceback__)])
+        # itertools.chain is in an extension module and may be unavailable
+        for it in its:
+            for x in it:
+                yield x
```

Comment:
This looks very similar to [traceback._iter_chain() from Python 3.4](https://github.com/python/cpython/blob/3.4/Lib/traceback.py#L120-L138), but differs in some small details.   Where did it come from?